c/misra/src/rules/RULE-12-6/AtomicAggregateObjectDirectlyAccessed.ql

```diff
@@ -0,0 +1,35 @@
+/**
+ * @id c/misra/atomic-aggregate-object-directly-accessed
+ * @name RULE-12-6: Structure and union members of atomic objects shall not be directly accessed
+ * @description Accessing a member of an atomic structure or union results in undefined behavior.
+ * @kind problem
+ * @precision very-high
+ * @problem.severity error
+ * @tags external/misra/id/rule-12-6
+ *       external/misra/c/2012/amendment4
+ *       correctness
+ *       concurrency
+ *       external/misra/obligation/required
+ */
+
+import cpp
+import codingstandards.c.misra
+
+from Expr expr, Field field
+where
+  not isExcluded(expr, Concurrency6Package::atomicAggregateObjectDirectlyAccessedQuery()) and
+  not expr.isUnevaluated() and
+  (
+    exists(FieldAccess fa |
+      expr = fa and
+      fa.getQualifier().getUnderlyingType().hasSpecifier("atomic") and
```

Comment:
Technically you don't need the getUnderlyingType here

Reply:
Would it not be possible to have, e.g.,

```c
typedef atomic struct { ... } my_atomic_struct;

my_atomic_struct foo;
foo->bar
```

which would require `getUnderlyingType` to detect?

Reply:
I see `hasSpecifier` recursively checks through typedefs, confirmed working with a test, this can indeed be removed. Thanks!